### PR TITLE
Clean up after PR#6

### PR DESCRIPTION
**Pull request by andresrc**
Follow up to https://github.com/jenkinsci/scm-api-plugin/pull/6

That PR upgrades the baseline but did not address `TODO` comments pending on such upgrade.

This PR fixes this. Some small additional cleanups (substitute deprecated findbugs annotations) are also included in the modified files.

Thanks @stephenc and @jglick for comments on the previous PR.

@reviewbybees

### CODE REVIEW

**Comment by armfergom** [comment]
Looking at comments in previous PR (#6), this LGTM :bee:

File: src/main/java/jenkins/scm/api/SCMHead.java [hunk 1]
```diff
@@ -38,12 +36,14 @@
...
 import java.util.logging.Level;
 import java.util.logging.Logger;
-import jenkins.model.Jenkins;
 import jenkins.model.TransientActionFactory;
 import jenkins.scm.api.actions.ChangeRequestAction;
 import org.kohsuke.stapler.export.Exported;
 import org.kohsuke.stapler.export.ExportedBean;
 
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
```
Comment (jglick) [inline src/main/java/jenkins/scm/api/SCMHead.java]: I suspect @stephenc will bug this.

File: src/main/java/jenkins/scm/api/SCMHead.java [hunk 2]
```diff
@@ -139,19 +139,16 @@ public String toString() {
...
-        Jenkins j = Jenkins.getInstance(); // TODO 1.572+ ExtensionList.lookup
-        if (j != null) {
-            for (TransientActionFactory<?> taf : j.getExtensionList(TransientActionFactory.class)) {
-                if (taf.type().isInstance(this)) {
-                    try {
-                        actions.addAll(createFor(taf));
-                    } catch (Exception e) {
-                        LOGGER.log(Level.SEVERE, "Could not load actions from " + taf + " for " + this, e);
-                    }
+        for (TransientActionFactory<?> taf : ExtensionList.lookup(TransientActionFactory.class)) {
```
Comment (jglick) [inline src/main/java/jenkins/scm/api/SCMHead.java]: Thanks, this was actually from my PR, filed before yours was merged IIRC.

File: src/main/java/jenkins/scm/api/SCMHead.java (hunk 1 above)
Comment (stephenc) [inline src/main/java/jenkins/scm/api/SCMHead.java]: :bug:

**Comment by stephenc** [comment]
JSR-305 is dormant, has been for a long while and shows no hope of ever producing an agreed set of annotations. Further more these annotations use javax. packages which it is not possible to use according to the Oracle Java binary licence, so applications can not use and ship these dependencies along with a JRE without violating the Oracle agreement.

> F. JAVA TECHNOLOGY RESTRICTIONS. You may not create, modify, or change the behavior of, or authorize your licensees to create, modify, or change the behavior of, classes, interfaces, or subpackages that are in any way identified as "java", "javax", "sun", “oracle” or similar convention as specified by Oracle in any naming convention designation.

The JSR-305 group has not defined any official releases according to its JSR page so the only implementations is a seemingly random implementation provided by the FindBugs team. Even if the Findbugs team were experts on the JSR (which some where) they are not official as there has been no vote and are not available from the JSR hompage - so the javax package name restriction still applies.

----

By using `javax` annotations bundled in jenkins.war we basically break the Oracle license in at least the docker image for Jenkins.

:bug:

BTW I am going to be filing a PR against Jenkins core that switches every annotation to a safe variant, e.g. the `edu.umd` one

**Comment by andresrc** [comment]
@stephenc is there a safe non-deprecated variant?

**Comment by stephenc** [comment]
Ignore the `@Deprecated`, there is [work on reversing that *or* getting the JSR revived](https://sourceforge.net/p/findbugs/bugs/1398/) /cc @jtnord

**Comment by andresrc** [comment]
@stephenc Addressed and rebased.

**Comment by andresrc** [comment]
@reviewbybees again esp. @stephenc

File: src/main/java/jenkins/scm/api/SCMHead.java (hunk 2 above)
Comment (amuniz) [inline src/main/java/jenkins/scm/api/SCMHead.java]: Why is this change here?

Comment (andresrc) [inline src/main/java/jenkins/scm/api/SCMHead.java]: I was cleaning up `TODO`s and we are now 1.572+

Comment (amuniz) [inline src/main/java/jenkins/scm/api/SCMHead.java]: Ok, missed the comment.

**Comment by jglick** [comment]
:bee:

**Comment by recena** [comment]
@andresrc Maybe this should be included in `scm-api-1.2`, my mistake /cc @stephenc
